### ISSUE TITLE
Give all magnetic field maps symmetric colorbar limits?

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#5828]
In https://github.com/sunpy/sunpy/pull/5825 I've set MDI and HMI maps to have symmetric colorbar limits about zero, but only for magnetic field (ie. not intensity) maps. This raises the question, **should we do this for all maps that have a magnetic field unit**, regardless of their source? Thoughts?

**Comment by ayshih** [comment]
I say no.  It makes sense for the LOS component, but it wouldn't make sense for magnitude of the vector field.

**Comment by dstansby** [comment]
> I say no. It makes sense for the LOS component, but it wouldn't make sense for the magnitude of the vector field.

Agreed. What about adding `np.any(data < 0) and np.an(data > 0)` to the check so this is only done for maps with positive and negative data?

**Comment by wtbarnes** [comment]
Where would this check happen? Do you mean when constructing the default normalization on `GenericMap`? Since we construct a colormap normalization for each source individually, I'm just a bit confused about how we would generalize this to some subset of sources.
